src/main/java/com/mongodb/kafka/connect/source/topic/mapping/CustomDataTypeTopicMapper.java

```diff
@@ -33,30 +36,55 @@ public void configure(final MongoSourceConfig config) {
     this.prefix = prefix.isEmpty() ? prefix : prefix + separator;
     this.suffix = suffix.isEmpty() ? suffix : separator + suffix;
     this.database = config.getString(DATABASE_CONFIG);
-    this.collection = config.getString(COLLECTION_CONFIG);
     this.dlqTopicName = config.getString(OVERRIDE_ERRORS_DEAD_LETTER_QUEUE_TOPIC_NAME_CONFIG);
   }
 
+  /**
+   * Dynamically builds and returns the topic name from changeStreamDocument param.
+   * In the case of operationType=INSERT, DATA_TYPE_FIELD field is located in the FULL_DOCUMENT_FIELD of the Document.
```

Comment:
In the case of operationType=INSERT and DATA_TYPE_FIELD field is located in the FULL_DOCUMENT_FIELD we will create a topic name includes the data type value.

Reply:
just to simplify the text in row #44...